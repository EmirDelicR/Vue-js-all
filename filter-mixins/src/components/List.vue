<template>
    <div>
        <h1>Mixins</h1>
        <hr/>
        <input v-model="filterText">
        <ul>
            <li v-for="(item, index) in filteredFruits" :key="index">{{ item }}</li>
        </ul>
    </div>

</template>

<script>
import { fruitMixin } from '../fruitMixin.js'

export default {
    name: 'List',
    mixins: [fruitMixin],
    created () {
        console.log("List component")
    }
}
</script>

<style>
</style>
