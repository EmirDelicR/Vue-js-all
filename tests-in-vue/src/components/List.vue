<template>
    <div id="some-id" class="list">
        {{ items }}
        <ul class="list-cl">
            <li v-for="(item, index) in items" :key="index">{{ item }}</li>
        </ul>
    </div>
</template>

<script>
export default {
    name: 'list',
    props: {
        items: {
            type: Array
        }
    }
}
</script>

<style>
</style>
