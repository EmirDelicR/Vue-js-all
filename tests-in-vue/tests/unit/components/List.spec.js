import { shallowMount } from '@vue/test-utils'
import List from '@/components/List.vue'

describe('List.vue', () => {
  let data
  const items = ['item 1', 'item 2']
  beforeEach(() => {
    data = {
      propsData: { items },
    }
  })

  it('testing component name', () => {
    const wrapper = shallowMount(List, data)
    expect(wrapper.name()).toMatch('list')
  })

  it('renders li for each item in props.items', () => {
    const wrapper = shallowMount(List, data)
    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(List, data)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('testing if wrapper have div, ul, li elements', () => {
    const wrapper = shallowMount(List, data)
    expect(wrapper.contains('div')).toBe(true)
    expect(wrapper.contains('ul')).toBe(true)
    expect(wrapper.contains('li')).toBe(true)
  })

  it('testing wrapper props', () => {
    const wrapper = shallowMount(List, data)
    expect(wrapper.props().items.length).toBe(2)
    expect(wrapper.props().items[0]).toMatch('item 1')
    expect(wrapper.props().items[1]).toMatch('item 2')
  })
})
