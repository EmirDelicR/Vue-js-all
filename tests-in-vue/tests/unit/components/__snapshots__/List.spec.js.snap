// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<div id="some-id" class="list">
  [
  "item 1",
  "item 2"
  ]
  <ul class="list-cl">
    <li>item 1</li>
    <li>item 2</li>
  </ul>
</div>
`;
